From: "Barak A. Pearlmutter" <barak+git@pearlmutter.net>
Date: Tue, 31 Jan 2017 13:44:35 +0000
Subject: autotools updates

- Misc autoupdate changes.
- Use AS_HELP_STRING.
- Remove obsolete autoconf macro STDC_HEADERS.
---
 Makefile.am  |  1 +
 configure.ac | 30 +++++++++++++-----------------
 2 files changed, 14 insertions(+), 17 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 2b7ef71..da11831 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -16,6 +16,7 @@
 ## along with this program; if not, write to the Free Software
 ## Foundation, Inc., 59 Temple Place, Suite 30, Boston, MA 02111-1307, USA.
 
+ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = include libemf src doc tests
 
 dist-hook:
diff --git a/configure.ac b/configure.ac
index 8b47398..067580d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2,25 +2,25 @@
 # Copyright (C) 2002, 2003 lignum Computing, Inc. <dallenbarnett@users.sourceforge.net>
 # $Id: configure.ac 98 2020-06-07 13:10:19Z dallenbarnett $
 
-AC_INIT([libEMF], 1.0.13, dallenbarnett@users.sourceforge.net)
+AC_INIT([libEMF],[1.0.13],[dallenbarnett@users.sourceforge.net])
 AC_CONFIG_AUX_DIR(config)
-AM_CONFIG_HEADER([config/config.h])
-AC_CONFIG_SRCDIR([include/libEMF/emf.h])
-AM_INIT_AUTOMAKE
+AC_CONFIG_SRCDIR([libemf/libemf.cpp])
+AC_CONFIG_HEADERS([config/config.h])
+AC_CONFIG_MACRO_DIRS([m4])
+AM_INIT_AUTOMAKE([-Wall])
 
 # Checks for programs.
 AC_PROG_CXX
 AC_PROG_CC
 AC_PROG_INSTALL
-#AC_PROG_LIBTOOL
 AC_PROG_MAKE_SET
+AM_PROG_AR
 
 LT_INIT
 
 # Checks for libraries.
 
 # Checks for header files.
-AC_HEADER_STDC
 AC_CHECK_HEADERS([errno.h stddef.h stdlib.h string.h unistd.h limits.h])
 
 # Checks for typedefs, structures, and compiler characteristics.
@@ -29,10 +29,7 @@ AC_C_INLINE
 AC_TYPE_SIZE_T
 
 # Checks for library functions.
-AC_FUNC_MALLOC
-AC_FUNC_MEMCMP
-AC_CHECK_LIB(m, ceil)
-AC_CHECK_FUNCS([floor memset select])
+AC_SEARCH_LIBS([ceil], [m])
 
 # Autoheader seems to complain about a lack of these (perhaps rightly)
 AH_TEMPLATE(LIBEMF_DEBUG, [Turn on any addition debugging edits])
@@ -42,7 +39,7 @@ AH_TEMPLATE(ENABLE_EDITING, [Enable the EditEnhMetaFile() function])
 # disable threads
 
 AC_ARG_ENABLE(debug,
-[  --enable-debug          enable libEMF debugging [default=no]],
+AS_HELP_STRING([--enable-debug],[enable libEMF debugging [default=no]]),
   enable_debug=$enableval, enable_debug=no)
 if test "x$enable_debug" = xyes; then
   AC_DEFINE(LIBEMF_DEBUG, 1)
@@ -56,7 +53,7 @@ your CFLAGS])
 fi
 
 AC_ARG_ENABLE(threads,
-[  --enable-threads           enable threads [default=yes]],
+AS_HELP_STRING([--enable-threads],[enable threads [default=yes]]),
   have_threads=$enableval, have_threads=yes)
 if test "x$have_threads" = xyes; then
 THREAD_LIBS=
@@ -65,13 +62,12 @@ AC_CHECK_HEADERS(pthread.h,
 
   save_LIBS=$LIBS
   LIBS="$LIBS -lpthread"
-  AC_TRY_LINK([#include <pthread.h>],[
-pthread_create (NULL, NULL, NULL, NULL)],[
+  AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <pthread.h>]], [[
+pthread_create (NULL, NULL, NULL, NULL)]])],[
     AC_MSG_RESULT(yes)
     CFLAGS="$CFLAGS -D_REENTRANT -DPTHREADS"
     CXXFLAGS="$CXXFLAGS -D_REENTRANT -DPTHREADS"
-    THREAD_LIBS="$THREAD_LIBS -lpthread"],
-    AC_MSG_RESULT(no)))
+    THREAD_LIBS="$THREAD_LIBS -lpthread"],[AC_MSG_RESULT(no)]))
   LIBS=$save_LIBS
 AC_SUBST(THREAD_LIBS)
 else
@@ -83,7 +79,7 @@ fi
 # Adds a lot of code and data to the library, though.
 
 AC_ARG_ENABLE(editing,
-[  --enable-editing     enable EditEnhMetaFile() function [default=no]],
+AS_HELP_STRING([--enable-editing],[enable EditEnhMetaFile() function [default=no]]),
   enable_editing=$enableval, enable_editing=no)
 if test "x$enable_editing" = xyes; then
    AC_DEFINE(ENABLE_EDITING, 1)
